### PR TITLE
Gameboard Update

### LINKED ISSUE #2: [Feature 2] Move System
A move system that allows players and computers to play legal moves. Moves interface is implemented by class Player.

### LINKED ISSUE #3: [Feature 4] User Interface
A package of classes that implements the graphical views of the game.

### LINKED ISSUE #19: [Feature 9] Interface adapters
To create controller class in order to maintain the clean architecture principle which will work with UI.

### PR DESCRIPTION
**Pull request by NonLan**
- The gameboard opens with all cards displayed correctly
- lastPlayed updates in UI
- Able to play cards (also reflected in UI)
- TODO: still need to make the game continue after a card is played/drawn by the real player

### CODE REVIEW

**Comment by NonLan** [comment]
Note: Some classes have been completely commented out because they have errors that prevent us from running GameManager. If you're able to fix them, please do!

File: src/main/java/UI/GameManager.java [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+
 import javax.swing.*;
+
 /**
  * the Main class that the users interact with
  */
 public class GameManager {
 
+
+    Controller c;
```
Comment (LogicalLunatic) [inline src/main/java/UI/GameManager.java:14]: Don't need Controller here

File: src/main/java/UI/RegularModeSetup.java [hunk 2]
```diff
@@ -149,15 +149,17 @@ public void actionPerformed(final ActionEvent e) {
         startGame.addActionListener(new ActionListener() {
             @Override
             public void actionPerformed(final ActionEvent e) {
-                c.startGame();
+                ui.generateGameBoard(c);
+                c.startGame(ui);
```
Comment (LogicalLunatic) [inline src/main/java/UI/RegularModeSetup.java:153]: May or may not be creating multiple windows. (Please look into)

**Review by SeanGrindal — COMMENTED** [review]
Looks mostly good but one thing, which doesn't immediately have to do with the PR, but is more of a general comment, is that we could improve the readability of the frontend code a lot. Tons of methods are called line by line with few comments about what's happening. 

I'd suggest breaking up these groups into smaller functions or add some more comments.

File: src/main/java/UI/StartPage.java [hunk 3]
```diff
@@ -16,70 +20,67 @@ public class StartPage {
     JButton teamButton;
     Controller c;
```
Comment (LogicalLunatic) [inline src/main/java/UI/StartPage.java:21]: Remove the local controller (The Controller declared in StartPage). Use the Controller from the super class (UIComponent).

**Review by LogicalLunatic — APPROVED** [review]

**Pull request merged**